src/NServiceBus.Core/Reliability/Outbox/OutboxMessage.cs

```diff
@@ -26,19 +36,6 @@ public OutboxMessage(string messageId)
         /// <summary>
         ///     The list of operations performed during the processing of the incoming message.
         /// </summary>
-        public List<TransportOperation> TransportOperations
-        {
-            get
-            {
-                if (transportOperations == null)
-                {
-                    transportOperations = new List<TransportOperation>();
-                }
-
-                return transportOperations;
-            }
-        }
-
-        List<TransportOperation> transportOperations;
+        public IEnumerable<TransportOperation> TransportOperations { get; private set; }
```

Comment:
since multiple operations are performed on this property it is a very dangerous API. depending on the specific IEnumerable returned we risk consuming the enumerable using things like `.Count()`. this should be changed back to a List